### PR TITLE
add more unit tests for the referenced secret

### PR DESCRIPTION
**Pull request by bmeng**
fix some small issues in the code and add more test coverage for the referenced secret.

### CODE REVIEW

File: pkg/controller/clusterdeployment/clusterdeployment_controller.go [hunk 1]
```diff
@@ -133,8 +133,10 @@ func (r *ReconcileClusterDeployment) Reconcile(request reconcile.Request) (recon
 		}
 	}
 
+	scName := config.PagerDutySecretName
```
Comment (cblecker) [inline pkg/controller/clusterdeployment/clusterdeployment_controller.go:136]: Do we need to reallocate this or can we just use it directly?

File: pkg/controller/clusterdeployment/clusterdeployment_controller_test.go [hunk 2]
```diff
@@ -508,3 +600,46 @@ func verifyNoConfigMapExists(c client.Client) bool {
...
+	}
+
+	return true
+}
+
+// verifyNoSecretExists verifies that the secret which referred by SyncSet does not exist
+func verifyNoSecretExists(c client.Client, expected *SecretEntry) bool {
+	secretList := &corev1.SecretList{}
+	opts := client.ListOptions{Namespace: testNamespace}
+	err := c.List(context.TODO(), &opts, secretList)
```
Comment (cblecker) [inline pkg/controller/clusterdeployment/clusterdeployment_controller_test.go:630]: Is there a reason we would do a List instead of a Get?

Comment (bmeng) [reply pkg/controller/clusterdeployment/clusterdeployment_controller_test.go:630]: Just followed the desigh for the syncset and the configmap which verify the non exists. It is better to keep them the same?

**Pull request merged**